mautrix_signal/portal.py

```diff
@@ -476,11 +484,41 @@ async def handle_matrix_reaction(
 
         # Signal doesn't seem to use variation selectors at all
         emoji = variation_selector.remove(emoji)
+        try:
+            retry_count = await self._signal_send_with_retries(
+                sender,
+                event_id,
+                send_fn=self._handle_matrix_reaction,
+                event_type=EventType.REACTION,
+                reacting_to=reacting_to,
+                emoji=emoji,
+            )
+        except Exception as e:
+            self.log.exception("Sending reaction failed")
+            sender.send_remote_checkpoint(
+                MessageSendCheckpointStatus.PERM_FAILURE,
+                event_id,
+                self.mxid,
+                EventType.REACTION,
+                error=e,
+            )
+        else:
+            sender.send_remote_checkpoint(
+                MessageSendCheckpointStatus.SUCCESS,
+                event_id,
+                self.mxid,
+                EventType.REACTION,
+                retry_num=retry_count,
+            )
+            await self._send_delivery_receipt(event_id)
 
+    async def _handle_matrix_reaction(
+        self, sender: u.User, event_id: EventID, reacting_to: EventID, emoji: str
+    ) -> None:
         message = await DBMessage.get_by_mxid(reacting_to, self.mxid)
         if not message:
             self.log.debug(f"Ignoring reaction to unknown event {reacting_to}")
-            return
+            raise NotConnected(f"Ignoring reaction to unknown event {reacting_to}")
```

Comment:
This looks like the catch in `_signal_send_with_retries` is going to log an unnecessary exception and send the checkpoint with a message that isn't true ("not connected to signald").

We should probably stop abusing existing exception classes and just make new ones when appropriate, a `class UnknownReactionTarget(Exception): pass` or something at the top of the file is fine. The handler can then catch both and adjust the log level and checkpoint message appropriately

Reply:
OK, that sounds good. I wasn't really loving this implementation, so I think that adding a new exception class makes sense here.